include/picongpu/plugins/misc/rangeToWindow.hpp

```diff
@@ -0,0 +1,115 @@
+/* Copyright 2017-2022 Rene Widera
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/simulation/control/Window.hpp"
+
+#include <pmacc/mappings/simulation/SubGrid.hpp>
+
+#include <algorithm>
+#include <string>
+
+
+namespace picongpu
+{
+    namespace plugins
+    {
+        namespace misc
+        {
+            /** Calculate window offsets and sizes based on a user given range description.
+             *
+             * @tparam T_dim number of dimensions the simulation is compiled for
+             * @param dataDomain PIConGPU data domain description
+             * @param inputWindow Selected window within the data domain e.g. the selected moving window.
+             * @param selectedRange String with range description for each dimension. Dimensions are separated by
+             *                      comma. The range for a dimension is of the form BEGIN:END where BEGIN is included
+             *                      and END excluded from the domain.
+             *                      If the begin or end of the range is outside of the data domain the values will be
+             *                      cropped to fit the data domain.
+             * @return Window description with local and global information.
+             */
+            template<uint32_t T_dim>
+            inline Window rangeToWindow(
```

Comment:
I believe this function actually does intersection of `inputWindow` and `selectedRange`, and not merely conversion of range to window. If so, perhaps a more direct naming could help

Reply:
Maybe we could call it `intersectRangeWithWindow` or `intersectRangeWindow` or something along these lines